Update state structure

DDKatch opened the pull request:
![2018-12-03 13 38 27](https://user-images.githubusercontent.com/14095146/49368817-d4f9c700-f700-11e8-987a-09e5222cc1ab.gif)

DDKatch committed:
Add card to card list on client side

DDKatch committed:
Try to move out ajax query logic into separate file

DDKatch commented:
@datarockets/cuties what do you think about [this?](https://github.com/datarockets/loopback-to-do/pull/14/commits/79ad4e3e56885278ce135d0bae36a294949bb46c)

DDKatch commented:
![](https://github.trello.services/images/mini-trello-icon.png) [When I add new todo card, I’d like to see it in todo list](https://trello.com/c/6HrXhXsl/28-when-i-add-new-todo-card-id-like-to-see-it-in-todo-list)

AleksSenkou commented on client/src/api/v1/cards/index.js:
```diff
@@ -0,0 +1,17 @@
+  },
+}
+
+export default api
```
@DDKatch I do like it 👍 

The only thing here - such requests should be placed near a container file, so:
```
AppContainer/
  index.js
  requests.js (or similar) P.S. there is a community name for such files, you may want to ask in #dev channel
```

And all requests should be described in this file
That's needed for keeping module system

AleksSenkou commented on client/src/index.js:
```diff
@@ -4,6 +4,6 @@ import 'font-awesome/css/font-awesome.min.css'
+import AppContainer from './containers/AppContainer'
 
-ReactDOM.render(<App />, document.getElementById('root'))
+ReactDOM.render(<AppContainer />, document.getElementById('root'))
```
👍

AleksSenkou commented on client/src/containers/AppContainer/index.js:
```diff
@@ -0,0 +1,34 @@
+class AppContainer extends Component {
+  state = {
+    todoCards: [],
+    todoCardForm: { description: '' },
```
@DDKatch why do we need `todoCardForm ` state here? not clear

AleksSenkou commented on client/src/components/App/index.js:
```diff
@@ -2,24 +2,28 @@ import React from 'react'
+    <div className="pt-4">
+      <div className="container-fluid">
+        <section>
+          <CreateCardFormContainer addTodoCard={props.addTodoCard} />
```
`onCreate` instead of `addTodoCard `

AleksSenkou commented on client/src/components/App/index.js:
```diff
@@ -2,24 +2,28 @@ import React from 'react'
+          <CreateCardFormContainer addTodoCard={props.addTodoCard} />
+        </section>
+        <section>
+          <TodoCardList todoCards={props.todoCards} />
```
`items` instead of `todoCards`

just how it always used

AleksSenkou reviewed (commented):
Looks good, here are several comments. I'd like to re-review after them

DDKatch committed:
Make db config more flexible

AleksSenkou commented on server/src/config/database.js:
```diff
@@ -0,0 +1,26 @@
+    host: '127.0.0.1',
+    dialect: 'postgres',
+  },
+}
```
❤️

DDKatch committed:
Remove warning

sequelize deprecated String based operators are now deprecated. Please use Symbol based operators for better security, read more at http://docs.sequelizejs.com/manual/tutorial/querying.html#operators node_modules/sequelize/lib/sequelize.js:242:13

DDKatch committed:
Add debug for backend

DDKatch committed:
Fix backend and frontend integration

nikitasakov commented:
I think that looks not bad, but I am not an expert in react 🙂. I myself need an example. 🙃

AleksSenkou commented on client/src/components/App/index.js:
```diff
@@ -2,24 +2,29 @@ import React from 'react'
 import CreateCardFormContainer from 'src/containers/CreateCardFormContainer'
-import TodoCardListContainer from 'src/containers/TodoCardListContainer'
+import TodoCardList from 'src/components/TodoCardList'
+import TodoCard from 'src/components/TodoCard'
```
@DDKatch when we are talking about List and items, we usually use such pattern:

```
TodoCardList/
  Item/
    index.js
  index.js
```

So, dependent Item is under List hierarchy because Item isn't used separately 😉

AleksSenkou commented on client/src/containers/AppContainer/index.js:
```diff
@@ -0,0 +1,31 @@
+      (response) => { this.setState({ todoCards: response }) },
+      (error) => {
+        alert(error)
+      },
```
```js
      (error) => {
        alert(error)
      },
```
can be done in one line, thoughts?

AleksSenkou commented on server/package.json:
```diff
@@ -6,6 +6,7 @@
   "private": true,
   "scripts": {
     "dev": "nodemon -w src --exec \"babel-node src\"",
+    "debug": "nodemon -w src src/index.js --exec 'babel-node --inspect'",
```
❤️

AleksSenkou replied:
Cool to have it, as for future: let's do infra changes in a separate task for easier review and testing 😉

DDKatch replied:
@AleksSenkou you're right, but I wasn't confident that I can fix bug without it, so added here 😉

AleksSenkou commented on server/src/models/index.js:
```diff
@@ -3,17 +3,10 @@ import path from 'path'
-} else {
-  sequelize = new Sequelize(config.database, config.username, config.password, config)
-}
+const sequelize = new Sequelize(config.database, config.username, config.password, config)
```
👍

AleksSenkou commented on client/src/containers/AppContainer/index.js:
```diff
@@ -0,0 +1,31 @@
+import App from 'src/components/App'
+import api from './requests'
+
+class AppContainer extends Component {
```
`AppContainer ` is a part of `containers` folder, so can be used without `Container` postfix.

The same pattern as for `components/`

DDKatch replied:
@AleksSenkou actually there'll be a conflict with `App` component if I rename it

AleksSenkou replied:
@DDKatch yeah, what about file naming? can we not use `Container` there?

DDKatch replied:
@AleksSenkou of course 😉

AleksSenkou reviewed (commented):
Overall looks great, some comments are left:

DDKatch commented:
@datarockets/cuties going to merge in 10 mins, is it ok?

AleksSenkou reviewed (approved):
LGTM
